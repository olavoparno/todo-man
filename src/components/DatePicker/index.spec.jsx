import React from 'react'
import { shallow } from 'enzyme'

import DateSelector from './index'

describe('DateSelector component', () => {
  let component
  let startDate
  beforeAll(() => {
    startDate = new Date('Mon Jul 22 2019 23:15:02 GMT-0300 (Brasilia Standard Time)')
    component = shallow(<DateSelector startDate={startDate} />)
  })
  it('should render correctly', () => {
    expect(component).toMatchSnapshot()
  })
  it('should return instance as null', () => {
    expect(component.instance()).toEqual(null)
  })
  it('should render with correct date', () => {
    expect(component.props().selected).toEqual(startDate)
  })
})
