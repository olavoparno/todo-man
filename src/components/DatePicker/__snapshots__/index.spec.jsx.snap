// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateSelector component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DateSelector
    startDate={2019-07-23T02:15:02.000Z}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "allowSameDay": false,
      "dateFormat": "Pp",
      "dateFormatCalendar": "LLLL yyyy",
      "disabled": false,
      "disabledKeyboardNavigation": false,
      "dropdownMode": "scroll",
      "inlineFocusSelectedMonth": false,
      "locale": "pt",
      "monthsShown": 1,
      "nextMonthButtonLabel": "Next month",
      "onBlur": [Function],
      "onChange": [Function],
      "onClickOutside": [Function],
      "onFocus": [Function],
      "onInputClick": [Function],
      "onInputError": [Function],
      "onKeyDown": [Function],
      "onMonthChange": [Function],
      "onSelect": [Function],
      "onYearChange": [Function],
      "preventOpenOnFocus": false,
      "previousMonthButtonLabel": "Previous Month",
      "readOnly": false,
      "renderDayContents": [Function],
      "selected": 2019-07-23T02:15:02.000Z,
      "shouldCloseOnSelect": true,
      "showMonthYearPicker": false,
      "showTimeInput": false,
      "showTimeSelect": true,
      "strictParsing": false,
      "timeCaption": "Time",
      "timeFormat": "p",
      "timeInputLabel": "Time",
      "timeIntervals": 15,
      "withPortal": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "allowSameDay": false,
        "dateFormat": "Pp",
        "dateFormatCalendar": "LLLL yyyy",
        "disabled": false,
        "disabledKeyboardNavigation": false,
        "dropdownMode": "scroll",
        "inlineFocusSelectedMonth": false,
        "locale": "pt",
        "monthsShown": 1,
        "nextMonthButtonLabel": "Next month",
        "onBlur": [Function],
        "onChange": [Function],
        "onClickOutside": [Function],
        "onFocus": [Function],
        "onInputClick": [Function],
        "onInputError": [Function],
        "onKeyDown": [Function],
        "onMonthChange": [Function],
        "onSelect": [Function],
        "onYearChange": [Function],
        "preventOpenOnFocus": false,
        "previousMonthButtonLabel": "Previous Month",
        "readOnly": false,
        "renderDayContents": [Function],
        "selected": 2019-07-23T02:15:02.000Z,
        "shouldCloseOnSelect": true,
        "showMonthYearPicker": false,
        "showTimeInput": false,
        "showTimeSelect": true,
        "strictParsing": false,
        "timeCaption": "Time",
        "timeFormat": "p",
        "timeInputLabel": "Time",
        "timeIntervals": 15,
        "withPortal": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
